test/Router/Zf2Test.php

```diff
@@ -164,7 +168,7 @@ public function testMatch()
 
         $result = $zf2Router->match($request);
         $this->assertInstanceOf('Zend\Expressive\Router\RouteResult', $result);
-        $this->assertEquals('/foo/GET', $result->getMatchedRouteName());
+        $this->assertEquals('/foo^GET/GET', $result->getMatchedRouteName());
```

Comment:
I'm thinking we may want to strip the child route from the returned matched route, as it's superfluous.